### DISCUSSION TITLE
CodeView widget for CTk

### DISCUSSION THREAD

**Discussion by Akascape** [TomSchimansky/CustomTkinter#2517] [Showcase]
I recently made this new **CTkCodeBox** widget for displaying code inside the textbox, very easy to use.

# CTkCodeBox
**Repository Link:** https://github.com/Akascape/CTkCodeBox

![image](https://github.com/user-attachments/assets/a22c6142-afc8-4239-840f-76e06ef7c668)

## Features
- Multiple language support
- Code syntax highlighting
- Multiple Themes
- Right-click copy-paste menu
- Line numbers on left side
- Fully customizable
- Easy to Use

### [<img alt="GitHub repo size" src="https://img.shields.io/github/repo-size/Akascape/CTkCodeBox?&color=white&label=Download%20Source%20Code&logo=Python&logoColor=yellow&style=for-the-badge"  width="400">](https://github.com/Akascape/CTkCodeBox/archive/refs/heads/main.zip)

**Download the source code, paste the `CTkCodeBox` folder in the directory where your program is present.**
<br> **Requirements:** Pygments and TkLineNums

## Simple Usage
```python
import customtkinter
from CTkCodeBox import *

root = customtkinter.CTk()

codebox = CTkCodeBox(root, language="python")
codebox.pack(padx=10, pady=10, expand=True,fill="both")

root.mainloop()
```

**Comment by dipeshSam** [comment]
Wow! That's great. ⭐ 

Regards.

**Reply by SarthakTools** [reply]
yep...
